src/main/java/io/pravega/connectors/flink/Pravega.java

```diff
@@ -0,0 +1,332 @@
+/**
+ * Copyright (c) 2017 Dell Inc., or its subsidiaries. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+package io.pravega.connectors.flink;
+
+import io.pravega.client.ClientConfig;
+import io.pravega.client.stream.Stream;
+import io.pravega.client.stream.StreamCut;
+import io.pravega.connectors.flink.util.StreamWithBoundaries;
+import org.apache.flink.api.common.serialization.DeserializationSchema;
+import org.apache.flink.api.common.serialization.SerializationSchema;
+import org.apache.flink.table.api.ValidationException;
+import org.apache.flink.table.descriptors.ConnectorDescriptor;
+import org.apache.flink.table.descriptors.DescriptorProperties;
+import org.apache.flink.types.Row;
+import org.apache.flink.util.Preconditions;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+
+import static org.apache.flink.table.descriptors.ConnectorDescriptorValidator.CONNECTOR_VERSION;
+
+/**
+ * Pravega connector descriptor.
+ */
+
+public class Pravega extends ConnectorDescriptor {
```

Comment:
Wondering if the name of this class is a bit too generic.

Reply:
It does feel a bit too generic, but depends on what other implementations are doing. It sounds like Kafka is calling the connector descriptor `Kafka`:

https://ci.apache.org/projects/flink/flink-docs-release-1.7/api/java/org/apache/flink/table/descriptors/Kafka.html

I suppose we should leave it as is to follow the same pattern.